community/kernel/src/main/java/org/neo4j/kernel/impl/locking/Locks.java

```diff
@@ -107,6 +109,9 @@ public Factory( String key, String... altKeys )
         /** Release all locks. */
         void releaseAll();
 
+        /** Associate with transaction. */
+        void setTx( Transaction tx );
```

Comment:
Actually, there seems to be a problem with that. The clients are pooled in forseti and even if we do not add the setTx to the Locks.Client interface, we would still need it on the ForsetiClient level, and the generalised flyweight pool would need a mechanism for abstractly initialising whatever it carries. Should be doable, but the change will be more complex.